src/i2c.h

```diff
@@ -3,10 +3,15 @@
 
 #include "stdint.h"
 
-uint8_t I2C_Write(uint8_t slave_addr, uint16_t reg_addr, uint32_t val, uint8_t addr16, uint8_t bnum);
-uint32_t I2C_Read(uint8_t slave_addr, uint16_t reg_addr, uint8_t addr16, uint8_t bnum);
+uint8_t I2C_Write8(uint8_t slave_addr, uint8_t reg_addr, uint8_t val);
+uint8_t I2C_Write8_Wait(uint16_t ms, uint8_t slave_addr, uint8_t reg_addr, uint8_t val);
+uint8_t I2C_Write16(uint8_t slave_addr, uint16_t reg_addr, uint16_t val);
+
+uint8_t I2C_Read8(uint8_t slave_addr, uint8_t reg_addr);
+uint8_t I2C_Read8_Wait(uint16_t ms, uint8_t slave_addr, uint8_t reg_addr);
```

Comment:
This is `C` not `C++`.
Check the function signature, the caller _can_ decide the wait time in with the `_Wait` variant.
I went for a separate function because that will be easier to refactor in the future - you can just search for that new function name. This is useful because its questionable why the wait is even required at some places, it might go away, or the time is reduced.
The benefit of having a function with integrated wait is that the compiler only needs to insert one `call` instruction (and instructions for pushing/popping the stack) vs two, reducing codesize.

Reply:
Thank you for the explanation, I see your point on how the separated implementation make refactor easer in the future.

Reply:
Thanks for the explanation, I value the point of making easer to refactor in the future.